app/javascript/miq-redux/middleware.ts

```diff
@@ -1,5 +1,10 @@
-import { Middleware } from 'redux';
+import thunk from 'redux-thunk';
+import { routerMiddleware } from 'connected-react-router';
 
-export const middlewares: Middleware[] = [];
-
-// add middleware for handling async operations and side effects
+/**
+ * @param history history for redux routing for react components * 
+ * @description
+ *  routerMiddleware adds middleware which is listening for location changes and updates router state
+ *  thunk is a middleware for async redux functions
+ */
+export const createMiddlewares = ({ history }) => ([routerMiddleware(history), thunk]);
```

Comment:
Can you add other middlewares here so we have a way to creating middlewares array?
```javascript
export const createMiddlewares = ({ history, middlewares }) => ([routerMiddleware(history), thunk, ...middlewares]);
```

Reply:
Sure i can do that